src/UrlSearchParams.ts

```diff
@@ -0,0 +1,151 @@
+import { duplicate } from './lang';
+
+/**
+ * A object with string keys and string or string array values that describes a query string.
+ */
+type ParamList = { [key: string]: string | string[] };
+
+/**
+ * Parse a query string, returning a ParamList object.
+ */
+function parseQueryString(input: string): ParamList {
+	const query = <{ [key: string]: string[] }> {};
+	input.split('&').forEach(entry => {
```

Comment:
Consider `for...of` here instead.  (If we were using `forEach`, we'd avoid fat arrow here since there's no context preservation going on, as per the style guide.)